frontend/src/pages/shopPage/shopPage.js

```diff
@@ -0,0 +1,73 @@
+import React, {useState, useEffect, useRef} from 'react';
+import CategoryList from '../../components/categoryList/categoryList';
+import SearchResult from '../../components/searchResult/searchResult';
+import {useParams, useLocation} from 'react-router-dom';
+import './shopPage.css'
+import productService from '../../services/product.service';
+import Button from '../../components/button/button';
+import PathBar from '../../components/pathBar/pathBar';
+
+function ShopPage() {
+    const param = useParams();
+    const [products, setProducts] = useState([])
+    const query = new URLSearchParams(useLocation().search);
+    const word = query.get("search");
+    const [count, setCount] = useState(9);
+    let hasMore = useRef(true)
+    let start = useRef(1);
+    let previous = useRef("")
+    let filter = "";
+    if (word) {
+        filter = word;
+    } else filter = param.category;
+
+    function showMore() {
+        start.current += 1;
+        setCount(9 * start.current)
+    }
+
+    useEffect(() => {
+        const testIfChanged = ((change) => {
```

Comment:
testIfChanged ? 
Please rename this variable so that it has better meaning.

Reply:
Do you have any suggestions how it should be named?